packages/@aws-cdk/aws-iotevents/lib/expression.ts

```diff
@@ -0,0 +1,75 @@
+import { IInput } from './input';
+
+/**
+ * Expression for events in Detector Model state
+ * @see https://docs.aws.amazon.com/iotevents/latest/developerguide/iotevents-expressions.html
+ */
+export abstract class Expression {
+  /**
+   * Create a expression from the given string
+   */
+  public static fromString(value: string): Expression {
+    return new StringExpression(value);
+  }
+
+  /**
+   * Create a expression for function `currentInput()`.
+   * It is evaluated to true if the specified input message was received.
+   */
+  public static currentInput(input: IInput): Expression {
+    return this.fromString(`currentInput("${input.inputName}")`);
+  }
```

Comment:
> add getCurrentInputCondition() to Input

I implement the feature `Input#getCurrentInputCondition()` in this `Expression` class.